flow/src/main/scala/org/alephium/flow/network/broker/InMemoryMisbehaviorStorage.scala

```diff
@@ -40,10 +40,11 @@ class InMemoryMisbehaviorStorage(val penaltyForgivness: Duration) extends Misbeh
   }
 
   def isBanned(peer: InetAddress): Boolean = {
+    val now = TimeStamp.now()
     get(peer) match {
       case Some(status) =>
         status match {
-          case Banned(_)     => true
+          case Banned(until) => now.isBefore(until)
```

Comment:
I'm not sure what it changes? 

 the `get` function is already removing addresses if the ban time as pass. The test is passing if we keep the previous logic

Reply:
`get(peer)` will return `None` as it get remove from the list 

https://github.com/alephium/alephium/pull/1279/files#diff-90c4a6ae0ff9164cae0c20ff5d1ae14e3efb2c05713373b1b315681c6a6fa6baR69-R71

Reply:
Sorry, my bad. We got some strange bans and thought this was an issue.

This actually shows some problems:
1. Deeply nested functional code is hard to maintain - better to write simple code.
2. We'd better rename `get` and `getAndExpire` to make it easier to understand.